tools/autograd/load_derivatives.py

```diff
@@ -334,90 +300,75 @@ def stride_expr(name):
         }),
     ]
 
-    for arg in args:
-        if 'name' not in arg:
-            # some returned arguments do not have names
-            continue
-
-        name = arg['name']
+    # find which arguments need to be saved
+    saved: List[Variable] = []
 
+    for name, type in zip(arg_names, arg_types):
         # First search the formula for expressions which can be evaluated
         # when the autograd Function is created to avoid saving variables
         for regex, info in REPLACEMENTS:
-            def repl(m):
-                suffix = info['suffix']
-                suffix = suffix(m) if callable(suffix) else suffix
-                expr = info['expr'](name) if 'expr' in info else m.group(0)
-                saved.append({
-                    'name': name + suffix,
-                    'type': info['type'],
-                    'expr': expr,
-                })
+            def repl(m: Match[str]) -> str:
+                suffix: str = info['suffix'](m) if callable(info['suffix']) else info['suffix']
+                expr: str = info['expr'](name) if 'expr' in info else m.group(0)
+                saved.append(Variable(
+                    name=name + suffix,
+                    type=info['type'],
+                    expr=expr,
+                ))
                 if 'res' in info:
-                    return info['res'](name)
+                    replacement: str = info['res'](name)
```

Comment:
Odd that you need a type annotation here

Reply:
`info['res']` is a lambda - mypy simply takes it as returning Any.